File: php/testcode/xxe_030.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_xxe_zip_xml_extract
function xxe030(BenchmarkRequest $req): BenchmarkResponse {
    $zipPath = '/var/app/uploads/' . basename($req->param('zip'));
    $zip = new ZipArchive();
    $zip->open($zipPath);
    $xml = $zip->getFromName('content.xml');
    $dom = new DOMDocument();
    $dom->loadXML($xml); // vuln-code-snippet vuln-line php_xxe_zip_xml_extract
    return BenchmarkResponse::ok($dom->saveXML());
}
// vuln-code-snippet end php_xxe_zip_xml_extract

File: ruby/testcode/xxe_030.rb
require_relative 'shared'
require 'ox'

# vuln-code-snippet start ruby_xxe_ox_permissive2
def parse_ox_permissive2_handler(req)
  xml = req.post('xml')
  doc = Ox.parse(xml) # vuln-code-snippet vuln-line ruby_xxe_ox_permissive2
  BenchmarkResponse.json({ root: doc.root.value })
end
# vuln-code-snippet end ruby_xxe_ox_permissive2
